fenixedu-ist-cms-components/src/main/java/pt/ist/fenixedu/cmscomponents/domain/homepage/HomepageListener.java

```diff
@@ -86,12 +86,14 @@ public static void createDefaultContents(Site newSite, Menu menu, User user) {
         newSite.setInitialPage(initialPage);
     }
     
-    private static void addContact(Person owner, Site homepageSite) {
+    private static void addContact(final Person owner, final Site homepageSite) {
         createWebAddress(owner, homepageSite.getFullUrl(), PartyContactType.INSTITUTIONAL, true);
     }
     
-    public static void updateName(UserProfile profile) {
-        Site site = profile.getPerson().getHomepage();
-        site.setName( new LocalizedString(I18N.getLocale(), profile.getDisplayName()));
+    public static void updateName(final UserProfile profile) {
+        final Site site = profile.getPerson().getHomepage();
```

Comment:
According to the reference bellow, a UserProfile may not have a reference to a Person, hence the getPerson method may return null, which will lead to a new NPE.

https://github.com/FenixEdu/fenixedu-academic/blob/master/src/main/dml/fenixedu-academic.dml#L4902-L4909